LiteCore/tests/N1QLParserTest.cc

```diff
@@ -117,6 +117,19 @@ TEST_CASE_METHOD(N1QLParserTest, "N1QL properties", "[Query][N1QL][C]") {
     CHECK(translate("SELECT db.*") == "{'WHAT':[['.db.']]}");
 
     CHECK(translate("select $var") == "{'WHAT':[['$var']]}");
+
+    // "custId" is implicitly scoped by the unique alias, "orders".
+    CHECK(translate("SELECT DISTINCT custId FROM orders where test_id = 'agg_func' ORDER BY custId") ==
+          "{'DISTINCT':true,'FROM':[{'AS':'orders'}],'ORDER_BY':[['.custId']],"
+          "'WHAT':[['.custId']],'WHERE':['=',['.test_id'],'agg_func']}");
+    try {
+        // "custId" must be explicitly scoped by a keyspace alias if joining with another table.
+        translate("SELECT custId, other.custId FROM orders JOIN orders other ON orders.test_id = other.test_id ORDER BY custId");
+        CHECK(false);
+    } catch (const exception& exc) {
+        string error = "property 'custId.' does not begin with a declared 'AS' alias";
+        CHECK(error == exc.what());
```

Comment:
FYI, the convention with CHECK / REQUIRE is to put the value to be tested first, then the expected value, like `CHECK(exc.what() == error);`. It doesn't really make a difference since both values get logged, but it's clearer to someone looking at a test failure.

Reply:
Understood, but the tricky thing is that the failure in translate() occurs at "string sql = parse(dict);", which simply throws. It passed the syntax check and succeeded in a valid JSON AST. When parsing that to SQL, it checks the AS alias and throws upon failure.

Reply:
There is actually a `CHECK_THROWS_WITH` function for checking that something throws with a given message.

Reply:
nice. go find it.